_posts/2020-08-05-#19160.md

```diff
@@ -7,17 +7,47 @@ authors: [ryanofsky]
+
+The [`IpcProcess`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L14-L34) class has [`spawn(exe_name, pid)`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L24-L26) and [`serve(exit_status)`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L31-L33) methods and is responsible for spawning new child processes, creating pipes child and parent processes can use to communicate, and passing pipe file descriptors to [`IpcProtocol`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L36-L72) objects in parent and child processes.
+
+The [`IpcProtocol`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L36-L72) class is what actually sends method calls across the pipe, turning every method call into a request and a response, and tracking object lifetimes. The `IpcProcotol` and `IpcProcess` classes could have been melded together into a single class, but separating them allows the spawn and pipe setup code to work with protocols other than [Cap'n Proto](https://capnproto.org/), which is the internal protocol currently used by the `IpcProtocol` class.
+
+The [`Init`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/init.h#L29-L61) interface is similar to other cross-process C++ interfaces like [`interfaces::Node`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/node.h#L45-L276), [`interfaces::Wallet`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/wallet.h#L51-L304), [`interfaces::Chain`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/chain.h#L60-L276) and [`interfaces::ChainClient`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc.120/src/interfaces/chain.h#L278-L309), providing virtual methods that can be called from other processes. What makes it special is that unlike other interfaces which are not implemented by every process—`interfaces::Node` is only implemented by the node process and `interfaces::Wallet` is only implemented by the wallet process—`interfaces::Init` is implemented by every process that supports being spawned, and is the initial interface returned by the [`IpcProtocol::connect(fd)`](https://github.com/ryanofsky/bitcoin/blob/pr/ipc-echo.7/src/interfaces/ipc.h#L50-L59) method, allowing the parent process to control the child process after the connection is established. The `interfaces::Init` interface has methods that allow the parent process get access to every interface supported by the child process, and when the parent process frees the `interfaces::Init` object, the child process shuts down.
```

Comment:
s/, and is/, and it is/ (missing subject)

maybe s/process get access/process to get access/